types/logger/logger.go

```diff
@@ -42,3 +46,61 @@ func (w funcWriter) Write(p []byte) (int, error) {
 
 // Discard is a Logf that throws away the logs given to it.
 func Discard(string, ...interface{}) {}
+
+type limitData struct {
+	lim          *rate.Limiter
+	lastAccessed time.Time
+	msgBlocked   bool
+}
+
+// RateLimitedFn implements rate limiting by fstring on a given Logf.
+// Messages are allowed through at a maximum of r messages/second, in
+// bursts of up to b messages at a time.
+func RateLimitedFn(logf Logf, f float64, b int) Logf {
+	r := rate.Limit(f)
+	msgList := make(map[string]limitData)
+	lastPurge := time.Now()
+
+	rlLogf := func(s string, args ...interface{}) {
```

Comment:
just return this directly?

and name `s` as `format`? (ala https://golang.org/pkg/log/#Logger.Printf)